### PR TITLE
[EN-5156] Implemented deeplink generator for Station's reports

### PR DESCRIPTION
**Pull request by marcos-vivar**
#### `Resolves`:
- [EN-5156](https://trustar.atlassian.net/browse/EN-5156) **Create get_deeplink method for reports**

#### `Brief Description`:

- **WHAT**: Added a deeplink generator for Station's reports, regarding its base URL.

- **WHY**: In order to have a permalink, concerning any given report, it is required to have a generator that accomplishes it.

- **HOW**: Created a method on `report_client` that constructs a permalink using a Report or a report ID.

### CODE REVIEW

File: trustar/report_client.py [hunk 1]
```diff
@@ -565,3 +565,36 @@ def redact_report(self, title=None, report_body=None):
...
+        True
+        >>> isinstance(report, Report)
+        True
+        >>> isinstance(deeplink, str)
+        True
+        >>> print(deeplink)
+        """
+
+        # default to interal ID if report ID field is present
+        if isinstance(report, Report) and report.id:
```
Comment (YaPeL) [inline trustar/report_client.py:590]: Pythonic way would be to use a try catch:

```
try:
    report = report.id
except AttributeError:
    report_id = report

```

File: trustar/report_client.py [hunk 2]
```diff
@@ -565,3 +565,36 @@ def redact_report(self, title=None, report_body=None):
...
+            report_id = report.id
+        # else treat report ID as a string
+        elif isinstance(report, str) and report:
```
Comment (YaPeL) [inline trustar/report_client.py:593]: problem with this implementation is that you are leaving any object that has a __str__ method out

File: trustar/report_client.py [hunk 3]
```diff
@@ -565,3 +565,36 @@ def redact_report(self, title=None, report_body=None):
...
+            report_id = report
+        else:
+            raise Exception("Cannot generate deeplink without an internal ID.")
```
Comment (YaPeL) [inline trustar/report_client.py:596]: ValueError would make more sense, but I think this is not necessary

File: trustar/report_client.py [hunk 4]
```diff
@@ -565,3 +565,36 @@ def redact_report(self, title=None, report_body=None):
...
+
+        deeplink = "{}/constellation/reports/{}".format(self._client.station, report_id)
```
Comment (YaPeL) [inline trustar/report_client.py:598]: IF the report that you got cannot be transformed into a string this is where the exception will raise, no need to manually raise it

File: trustar/report_client.py (hunk 2 above)
Comment (marcos-vivar) [reply trustar/report_client.py:593]: @YaPeL How is this so? I may not be following you on this one.

**Comment by YaPeL** [comment]
LGTM!

**Comment by marcos-vivar** [comment]
> Thank you for making this Marcos! I think it's a good addition to have. My one thought is that we have different types of reports in Station that have slightly different ways that the links are generated. Normally that would probably mean that we would create some kind of generic method that classes like ReportClient, IndicatorClient, and PhishingTriageClient could inherit and implement, but I don't think our current inheritance hierarchy lends itself to that and it's out of scope for the ticket. Maybe it's something that could be added later if we feel like we need it. Nice work!

@jagrvargen Loved this feedback because it rings many bells! I think we should create a ticket in [EN-5073](https://trustar.atlassian.net/browse/EN-5156) **"All things Python SDK"** to start a discussion on this effort.

**Comment by jagrvargen** [comment]
@marcos-vivar I'm for it!

**Pull request merged**